static/js/components/SourcesInGcn.jsx

```diff
@@ -0,0 +1,267 @@
+
+import Dialog from "@mui/material/Dialog";
+import DialogContent from "@mui/material/DialogContent";
+import MuiDialogTitle from "@mui/material/DialogTitle";
+import IconButton from "@mui/material/IconButton";
+import EditIcon from "@mui/icons-material/Edit";
+import Close from "@mui/icons-material/Close";
+import Typography from "@mui/material/Typography";
+import grey from "@mui/material/colors/grey";
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+
+import { showNotification } from "baselayer/components/Notifications";
+
+import * as SourceInGcnAction from "../ducks/sourcesingcn";
+
+dayjs.extend(utc);
+
+const useStyles = makeStyles((theme) => ({
+  container: {
+    width: "100%",
+    overflow: "scroll",
+  },
+  eventTags: {
+    marginLeft: "0.5rem",
+    "& > div": {
+      margin: "0.25rem",
+      color: "white",
+      background: theme.palette.primary.main,
+    },
+  },
+  buttons: {
+    display: "flex",
+    flexDirection: "row",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  button: {
+    maxWidth: "1.2rem",
+  },
+  buttonIcon: {
+    maxWidth: "1.2rem",
+  },
+}));
+const dialogTitleStyles = (theme) => ({
+  root: {
+    margin: 0,
+    padding: theme.spacing(2),
+  },
+  title: {
+    marginRight: theme.spacing(2),
+  },
+  closeButton: {
+    position: "absolute",
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: grey[500],
+  },
+});
+
+const DialogTitle = withStyles(dialogTitleStyles)(
+  ({ children, classes, onClose }) => (
+    <MuiDialogTitle className={classes.root}>
+      <Typography variant="h6" className={classes.title}>
+        {children}
+      </Typography>
+      {onClose ? (
+        <IconButton
+          aria-label="close"
+          className={classes.closeButton}
+          onClick={onClose}
+        >
+          <Close />
+        </IconButton>
+      ) : null}
+    </MuiDialogTitle>
+  )
+);
+
+const SourcesInGCN = ({
+  dateobs,
+  localizationName,
+  sourceId,
+  startDate,
+  endDate,
+  localizationCumprob,
+  sourcesIdList,
+}) => {
+  const dispatch = useDispatch();
+  const classes = useStyles();
+  const { permissions } = useSelector((state) => state.profile);
+  const [open, setOpen] = useState(false);
+
+  const sourcesingcn = useSelector((state) => state.sourcesingcn.sourcesingcn);
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const findCurrentState = () => {
+    let state = "pending";
+    if (
+      sourcesingcn.filter((s) => s.obj_id === sourceId).length !== 0 &&
+      sourcesingcn?.length > 0
+    ) {
+      if (
+        sourcesingcn.filter((s) => s.obj_id === sourceId)[0]
+          ?.confirmed_or_rejected === true
+      ) {
+        state = "confirmed";
+      } else if (
+        sourcesingcn.filter((s) => s.obj_id === sourceId)[0]
+          .confirmed_or_rejected === false
+      ) {
+        state = "rejected";
+      }
+    }
+    return state;
+  };
+
+  const currentState = findCurrentState();
```

Comment:
Why do we need this to be a function? Can we just calculate the value and assign it?